### ISSUE TITLE
Make target `generate-go-apis` isn't running on an API change

### ISSUE THREAD

**Issue by richardcase** [kubernetes-sigs/cluster-api-provider-aws#3336]
/kind bug
/area api
/priority critical-urgent
/triage accepted
/help

**What steps did you take and what happened:**
I have made a change to the [API definition](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/exp/api/v1beta1/awsmanagedmachinepool_types.go) for `AWSManagedMachinePool`. I then run `make generate` and i would expect that the [CRD definition](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/config/crd/bases/infrastructure.cluster.x-k8s.io_awsmanagedmachinepools.yaml) for `AWSManagedMachinePool` to be updated....but its not.

Looking at the command output, i see that the make target [.build/generate-go-apis](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/Makefile#L202) is skipped:

![image](https://user-images.githubusercontent.com/198425/159441881-2816fd0a-7363-4fa3-b68b-9eaffd92f822.png)

**What did you expect to happen:**
I would expect the CRD definition to be updated.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-aws version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

**Comment by LucaLanziani** [comment]
I wouldn't mind working on this if it's ok with you.

**Comment by Ankitasw** [comment]
@LucaLanziani please go ahead and assign it to yourself.

**Comment by Ankitasw** [comment]
@richardcase are you sure that this is valid issue, as I tried adding a field to `AWSManagedMachinePool` and it works fine for me, adds the field into the CRDs.
Maybe can you give it a try after doing `make clean`?

**Comment by richardcase** [comment]
Thanks @Ankitasw.

I just tried `make clean` and that did fix the issue i was having. However, if i make another change and run `make generate` again it doesn't run the [.build/generate-go-apis](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/Makefile#L202) target.

We shouldn't need to do `make clean` between every run of this as i may be iterating on a change.

**Comment by Ankitasw** [comment]
Yes, I have faced this issue before, even I was not sure why every time we have to so `make clean`.

**Comment by LucaLanziani** [comment]
The PR above should fix the behavior, I tried changing the `API_SRCS` but that adds an unexpected side effect, I'll try to clean that up tomorrow.

**Comment by LucaLanziani** [comment]
The PR was reverted because it was making all following PRs failing. Can we re-open the issue please?

**Comment by Ankitasw** [comment]
/reopen
@LucaLanziani please go ahead and make another PR if fix is ready 😄
